pkg/build/builder/common.go

```diff
@@ -66,6 +66,25 @@ func buildInfo(build *api.Build) []KeyValue {
 	return kv
 }
 
+// updateBuildStatus updates the current status of the build only if there was
+// an error
+func updateBuildStatus(c client.BuildInterface, build *api.Build) {
+	// NOTE: if reason is empty there's nothing to update, do i need this check
+	// here?
+	if build.Status.Reason == "" {
+		return
+	}
+
+	switch build.Status.Reason {
+	case api.StatusReasonPostCommitHookFailed:
+		build.Status.Message = "failed because of post commit hook"
```

Comment:
> Tests can do string.Contains() and check for prefix, like strings.HasPrefix("post commit hook returned error")

yes but i want tests to do "strings.HasPrefix(buildapi.PostCommitFailureReason)" instead.

>  think about 1000 possible causes for message "git clone failed", comparing to: "git clone failed because connection timeout connecting to github.com"

if we want something more specific than "git clone failed" as the message text, then i suggest the constant:
"git clone failed due to: %v".

so it's still a constant, just a customizable constant, and we run it through fmt.Sprintf.